### PR TITLE
sip: Fix for linuxbrew

### PR DESCRIPTION
**Pull request by saketkc**

### CODE REVIEW

**Comment by saketkc** [comment]
Avoiding the single line change that would have been easier, since the empty space generated breaks to `configure.py` script.

File: Library/Formula/sip.rb [hunk 1]
```diff
@@ -30,12 +30,15 @@ def install
 
     Language::Python.each_python(build) do |python, version|
       # Note the binary `sip` is the same for python 2.x and 3.x
-      system python, "configure.py",
-                     "--deployment-target=#{MacOS.version}",
```
Comment (sjackman) [inline Library/Formula/sip.rb]: Does  the following work? I'd prefer the smaller patch if it does.
```diff
- "--deployment-target=#{MacOS.version}",
+ ("--deployment-target=#{MacOS.version}" if OS.mac?),
````

Comment (saketkc) [inline Library/Formula/sip.rb]: That's what I tried. It still leads to an empty space, and `configure.py` recognises it as an extra option.

**Comment by sjackman** [comment]
Thanks for your contribution, Saket.

You can use `compact` to remove the `nil` entry from the vector. See here: https://github.com/Homebrew/linuxbrew/commit/0c2c2cfc7c0c85695d80debfec5220d2de527cd6

**Pull request closed without merging**